### PR TITLE
Added Vanilla, Denoising and Sparse AE and updated README.md

### PR DESCRIPTION
**Pull request by mathurpulkit**
Have added the code, trained model, outputs and separate README files each for vanilla and denoising Autoencoders.
Also modified main README.md file a bit.
We have now also added code for sparse autoencoders using L1 loss and KLDiv loss.
We have added labels to graph axes, added loss and hyperparameter details in the READMEs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Variational-DL
-### Contributors 
+
+#### Variational Deep Learning is a method of deep learning where we use Neural Networks to generate data, instead of drawing conclusions from it.
```
Comment (rishika2110) [inline README.md]: Don't put H4 for the content part. Please do the changes in both the readmes.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+
+#### We have currrently implemented two Autoencoders:
```
Comment (rishika2110) [inline README.md]: Use checkboxes to list out all the AEs which are going to be uploaded and tick them once uploaded.

Comment (rishika2110) [reply README.md]: Refer to NLP repository

File: denoise/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Denoising Autoencoder
+
+#### A Denoising Autoencoder is an autoencoder which learns to reproduce the input from a compressed representation of the output. The main difference between vanilla and denoisisng autoencoder is that a denoising autoencoder is trained on a noisy original input. This noise can be of any type(for eg: Gaussian Noise). The encoder neural network is trained to ignore the noise and create an efficient representation while the decoder tries to reconstruct the output.
```
Comment (rishika2110) [inline denoise/README.md]: Denoising AE isn't used for reproducing the input.

File: vanilla/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Vanilla Autoencoder
+
+#### A Vanilla Autoencoder is an autoencoder which learns to reproduce the input from a compressed representation of the output.
```
Comment (rishika2110) [inline vanilla/README.md]: It shouldn't be "compressed representation of output". Change it to "the compressed representation of the output of encoder i.e. encoding" or "latent space representation of the input obtained by passing through the encoder".

File: README.md (hunk 1 above)
Comment (mathurpulkit) [reply README.md]: Resolved the issue in all files.

File: README.md (hunk 2 above)
Comment (mathurpulkit) [reply README.md]: Added the checkboxes

File: denoise/README.md (hunk 3 above)
Comment (mathurpulkit) [reply denoise/README.md]: Rewrote the whole definition again.

File: vanilla/README.md (hunk 4 above)
Comment (mathurpulkit) [reply vanilla/README.md]: Made the suggested changes.

**Pull request merged**